framework/src/outputs/Output.C

```diff
@@ -156,6 +161,17 @@ Output::Output(const InputParameters & parameters)
     for (auto & me : add)
       _execute_on.push_back(me);
   }
+
+  if (isParamValid("output_limiting_function"))
+  {
+    const Function & olf = getFunction("output_limiting_function");
+    const PiecewiseBase * pwb_olf = dynamic_cast<const PiecewiseBase *>(&olf);
+    if (pwb_olf == nullptr)
+      mooseError("Function muse have a piecewise base!");
```

Comment:
I think this code illustrates this design is flawed. It only works with specific types of functions. When I think of a `Function` I think of something that is conceptually continuous. So the idea of a continuous function dictating discrete output times doesn't seem right

Reply:
This approach is the same as is used in IterationAdaptiveDT.  Due to the fact that IterationAdaptiveDT guarantees a solution step at the times listed in a piecewise linear function, we want to get output at exactly those times without needing to type the times one by one, which is the only option offered now.

Reply:
The thing is that I think the limiting function in `IterationAdaptiveDT` should be changed to allow general `Functions`. You could write a ParsedFunction for what you want your max dt to be for a given time for example

Reply:
Maybe there is some confusion about what these `Functions` are providing.  In our use, they are giving us the points in time when a solve must occur.  These times are the time values in the piecewise functions.  They are not specifying the dt.  The dt may be adjusted to hit the times from the functions, of course.

Reply:
I find the function use in `IterationAdaptiveDT` confusing 😄 That class has become a monster and we are trying to make it less important or just do less with the new time step compositing possible with @MengnanLi91 and @GiudGiud's work. 

So I would prefer the design of `IterationAdaptiveDT` not be a template for other design decisions. I think that something like what @GiudGiud proposed in #24696 would be much better.

Reply:
How long will @GiudGiud's solution take?  We can limp along with Joseph's fix local in a submodule, but that's not ideal.

Reply:
Not long I ll try my best

Reply:
@jasondhales a short-term solution I could live with is adding this new parameter as a deprecated parameter to indicate to users that this is not something they should rely on existing in the future